gcloud/pubsub/test_subscription.py

```diff
@@ -620,29 +603,68 @@ def test_check_iam_permissions_w_alternate_client(self):
 
         self.assertEqual(len(allowed), 0)
         self.assertEqual(len(conn1._requested), 0)
-        self.assertEqual(len(conn2._requested), 1)
-        req = conn2._requested[0]
-        self.assertEqual(req['method'], 'POST')
-        self.assertEqual(req['path'], PATH)
-        self.assertEqual(req['data'], REQUESTED)
+        self.assertEqual(len(conn2._requested), 0)
+        self.assertEqual(conn2._tested_iam_permissions,
+                         (self.SUB_PATH, ROLES))
 
 
-class _Connection(object):
+class _Connection(object):  # pylint: disable=too-many-instance-attributes
```

Comment:
I'm still *really* unhappy with tweaking the artificial numerical limits in the global config every time we run up against one of them:

- If the numbers are actually meaningful (almost never), then we should enforce them.
- If the numbers are "more guidelines, really", then disabling them locally allows them to be as useful as they were when we come across a solitary case where the guideline is deemed un-useful.
- If they are basically random opinions pulled from the air at Logilab, then we should disable them altogether.

In this case, I think the "too many attributes" bit is not a terrible guideline, but needs to be conditioned by the fact that we map APIs not under our control:  adoptiong the "don't share testing state between unrelated methods" strategy for the mock connections means the limit isn't helpful.